src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -20,15 +20,48 @@
 
 namespace Stratis.Bitcoin.Features.Miner
 {
+    /// <summary>
+    /// <see cref="PosMinting"/> is used in order to generate new blocks. It involves a sort of lottery, similar to proof-of-work,
+    /// but the chances of winning this lottery is proportional to how many coins you are staking, not on hashing power.
+    /// </summary>
+    /// <remarks>
+    /// Staking is attempted only if the node is fully synchronized and connected to the network.
+    /// If not it will wait  till node is synced. Only transactions that were confirmed at least
+    /// <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks ago are eligible for staking.
+    /// </remarks>
+    /// <remarks>
+    /// The overall process for "attempting" to mine a PoS block looks like this:
+    /// 1) Create new block with transactions from <see cref="TxMempool"/>
+    /// 2) Get UTXOs (Unspent Transaction Output) that can participate in staking (have suitable depth).
+    /// 3) Split those UTXOs in subsets so each subset contains max of <see cref="utxoStakeDescriptionsPerCoinstakeWorker"/> transactions
+    /// and run CoinstakeWorker for each subset. Splitting is done only to achieve a good level of parallelism.
```

Comment:
`<see>` for CoinstakeWorker 

OR (imho better) abstract the reader of these details and just say something like
`Split these UTXOs in subsets and create tasks processing each subset to allow for parallel processing.`

i think it is better because mentioning the actual constant we have there, that's a detail that maybe does not belong to the big picture, if one is interested in such a detail, he would go to the code and will find this constant and if the constant is properly described, the information will be there.